### PR TITLE
Addresses #53

### PR DESCRIPTION
**Pull request by imuchnik**
Adds extendedEmail block

### CODE REVIEW

File: src/main/groovy/jenkins/automation/utils/CommonUtils.groovy [hunk 1]
```diff
@@ -19,4 +24,31 @@ class CommonUtils {
...
+ * @param triggers List<String> triggers E.g Failure, Fixed etc...
+ * @param String emails Comma separated string of emails
+ * @param sendToDevelopers Default false,
+ * @param includeCulprits Default false,
+ * @param sendToRecipientList Default true
+ *
+ * @see <a href="https://github.com/cfpb/jenkins-automation/blob/gh-pages/docs/examples.md#common-utils" target="_blank">BDD job Example</a>
+ */
+
+    static void extendedEmail(context, List<String> triggers = ["Failure", "Fixed"], String emails, sendToDevelopers = false, includeCulprits = false, sendToRecipientList = true) {
```
Comment (marcesher) [inline src/main/groovy/jenkins/automation/utils/CommonUtils.groovy]: Should emails be a `List<String>` instead, to retain compatibility with the Builders? I don't have a strong opinion, but it seems to me that we should remain consistent

File: docs/examples.md [hunk 2]
```diff
@@ -179,3 +179,28 @@ job('test') {
...
+ job('example'){
+    CommonUtils.addDefaults(delegate)
+    }
+```
+### Extended Email
+
+import jenkins.automation.utils.CommonUtils
+
+ job('example'){
+   extendedEmail(delegate,
```
Comment (marcesher) [inline docs/examples.md]: This looks wrong. Wouldn't it be CommonUtils.extendedEmail(....)?

File: src/main/groovy/jenkins/automation/utils/CommonUtils.groovy (hunk 1 above)
Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/CommonUtils.groovy]: The extendedEmail block takes a string http://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.helpers.publisher.PublisherContext.extendedEmail and I am a little hesitant to diverge from that. I can still make it work for the builder doing cool groovy things http://grails.asia/groovy-list-to-string inside the builders themselves. **OR** I can override the function to also take in a list with a signature like so: +    static void extendedEmail(context, List<String> triggers = ["Failure", "Fixed"], List emails, sendToDevelopers = false, includeCulprits = false, sendToRecipientList = true)

File: docs/examples.md (hunk 2 above)
Comment (imuchnik) [inline docs/examples.md]: Oh why, yes! Yes, indeed!

**Comment by imuchnik** [comment]
@marcesher comments addressed.

File: example-jobs/checkmarx-security-job-example.groovy [hunk 3]
```diff
@@ -10,7 +10,7 @@ new CheckmarxSecurityJobBuilder(
         username: "sample-username",
         password: "sample-password",
         groupId: "",
-        scanRepo: "https://github.com/OrlandoSoto/ckan-browser-tests",
+        scanRepo:[[url: "https://github.com/OrlandoSoto/ckan-browser-tests"]],
```
Comment (marcesher) [inline example-jobs/checkmarx-security-job-example.groovy:13]: What's this? Looks like an unrelated change, and an incorrect one

Comment (imuchnik) [inline example-jobs/checkmarx-security-job-example.groovy:13]: No, it is correct. The builder was changed to accept a list of repos to scan. See #51.

File: src/main/groovy/jenkins/automation/utils/CommonUtils.groovy (hunk 1 above)
Comment (marcesher) [inline src/main/groovy/jenkins/automation/utils/CommonUtils.groovy]: ah, well then. Here are my 2 cents:

I think we should change the API to support a string, and only a string.

Yes, there will be some work for us to change any existing configs, but it's a few minutes of work and probably not a big deal.

I'll merge this version in now, and then we can plan a time -- probably next week -- with all interested parties to make the cutover from the List to a string.

I think we should be trying to write a) as little code as possible, b) code that doesn't deviate from job-dsl for no good reason

File: example-jobs/checkmarx-security-job-example.groovy (hunk 3 above)
Comment (marcesher) [inline example-jobs/checkmarx-security-job-example.groovy:13]: ah, I see, this is for #51 

thanks.
